src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeDistribution.java

```diff
@@ -0,0 +1,205 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.distribution.AbstractRealDistribution;
+import org.apache.commons.math3.distribution.LogNormalDistribution;
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.distribution.RealDistribution;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.spark.sv.evidence.ReadMetadata;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Holds the information characterizing and insert size distribution.
+ */
+public class InsertSizeDistribution implements Serializable {
+
+    private static final long serialVersionUID = -1L;
+
+    @VisibleForTesting
+    static final Type[] SUPPORTED_TYPES = { new NormalType(), new LogNormalType() };
+
+    public interface Type {
+        List<String> getNames();
+
+        AbstractRealDistribution fromMeanAndStdDeviation(final double mean, final double stddev);
+
+        default AbstractRealDistribution fromReadMetadataFile(final String meanString) {
```

Comment:
It seems rather fragile to parse a text file for this information.  We already have a means of serializing the ReadMetadata to a file.  Why don't you just read that?
You could change the ReadMetadata code to always write read-metadata.txt as well as read-metadata.bin.